### ISSUE TITLE
[7.0.4] Type defaultViewport can't be null (to take all window space)

### ISSUE THREAD

**Issue by spinlud** [puppeteer/puppeteer#6885]
* Puppeteer version: 7.0.4
* Node.js version: 12.x

Usually, to make viewport taking all the available space in window we need to set `defaultViewport` to null:

```js
const puppeteer = require('puppeteer');

(async () => {
    const windowWidth = 1472;
    const windowHeight = 828;
    
    const browser = await puppeteer.launch({
        headless: false,
        args: [
            "--enable-automation",
            "--start-maximized",
            `--window-size=${windowWidth},${windowHeight}`,
            "--no-sandbox",
            "--disable-setuid-sandbox",
            "--disable-dev-shm-usage",
            "--proxy-server='direct://",
            "--proxy-bypass-list=*",
            "--disable-gpu",
            "--disable-accelerated-2d-canvas",
        ],
        defaultViewport: null,
        pipe: true,
    });

    const page = (await browser.pages())[0];
    const url = 'https://www.blizzard.com/en-us/';

    await page.goto(url, {
        waitUntil: 'load',
    });
})();
```

This doesn't seem to be allowed in typescript (`defaultViewport` must be of type `Viewport` or `undefined`):

![image](https://user-images.githubusercontent.com/12783208/107784399-b0bbbb00-6d4b-11eb-807d-414876e92214.png)

But leaving it undefined does not give the desired result.

**Comment by Niek** [comment]
Can confirm the same issue.

**Comment by jef** [comment]
Remove `--window-size=${windowWidth},${windowHeight}` and `defaultViewport` in favor of `--start-maximized`.

**Comment by christian-bromann** [comment]
Proposed a fix for this in #6942
